cpp/src/io/parquet/page_enc.cu

```diff
@@ -127,7 +127,15 @@ __global__ void __launch_bounds__(512) gpuInitPageFragments(PageFragment *frag,
   dtype_len = (dtype == INT64 || dtype == DOUBLE) ? 8 : (dtype == BOOLEAN) ? 1 : 4;
   if (dtype == INT32) {
     uint32_t converted_type = s->col.converted_type;
-    dtype_len_in            = (converted_type == INT_8) ? 1 : (converted_type == INT_16) ? 2 : 4;
+    dtype_len_in            = [converted_type]() -> uint32_t {
```

Comment:
Might be better to define a function instead of this lambda, since the code is reused.